token-lending/program/src/processor.rs

```diff
@@ -2261,6 +2267,32 @@ fn get_switchboard_price(
     Decimal::from(price).try_div(price_quotient)
 }
 
+fn get_switchboard_price_v2(
+    switchboard_feed_info: &AccountInfo,
+    clock: &Clock,
+) -> Result<Decimal, ProgramError> {
+    const STALE_AFTER_SLOTS_ELAPSED: u64 = 240;
+
+    let feed = AggregatorAccountData::new(switchboard_feed_info)?;
+    let slots_elapsed = clock
+        .slot
+        .checked_sub(feed.latest_confirmed_round.round_open_slot)
+        .ok_or(LendingError::MathOverflow)?;
+    if slots_elapsed >= STALE_AFTER_SLOTS_ELAPSED {
+        msg!("Switchboard oracle price is stale");
+        return Err(LendingError::InvalidOracleConfig.into());
+    }
+
+    let price_float: f64 = feed.get_result()?.try_into()?;
+
+    // we just do this so we can parse coins with low usd value
+    // it might be better to just extract the mantissa and exponent from the float directly
+    let price_quotient = 10u64.pow(9);
+    let price = ((price_quotient as f64) * price_float) as u128;
+
+    Decimal::from(price).try_div(price_quotient)
```

Comment:
I think it's better to convert directly from `SwitchboardDecimal` to `math::Decimal` because typically programs don't use floating point math due to the [extra compute costs](https://docs.solana.com/developing/on-chain-programs/overview#float-support). In this case, I think you just need to multiply the mantissa by `18 (wad) - feed_result.scale`

Reply:
yeh i dont disagree, reasoning for this was just to have lowest amount of different code (switchboard v1 only had floats for a while so this code is copied from the same way we parse there). but i guess that is probably over cautious. i don't think the compute is much of an issue here (and if anything math::Decimal is not super efficient either)

Reply:
interestingly i switched to putting it in the decimal semi directly, and compute was exact same (lucky i guess?)